### PR TITLE
MLDB-1267 bucketize latest timestamp

### PR DESCRIPTION
**Pull request by jeremybarnes**
Causes the latest timestamp for the order by column to be used by the bucketize procedure.

Also fixes a bug in CSV timestamps that caused the test to fail originally.

@FinchPowers FYI

### CODE REVIEW

File: plugins/bucketize_procedure.cc [hunk 1]
```diff
@@ -122,10 +122,28 @@ run(const ProcedureRunConfig & run,
     SelectExpression select(SelectExpression::parse("1"));
     vector<shared_ptr<SqlExpression> > calc;
 
+    // We calculate an expression with the timestamp of the order by
+    // clause.  First, we need to calculate each of the order by clauses
+    for (auto & c: procedureConfig.inputData.stm->orderBy.clauses) {
+        cerr << "when clause for " << c.first->print() << endl;
```
Comment (steadtler) [inline plugins/bucketize_procedure.cc]: debug output?

Comment (jeremybarnes) [inline plugins/bucketize_procedure.cc]: thanks, fixed

**Comment by steadtler** [comment]
Looks fine besides the debug leftover + 1

File: testing/MLDB-1043-bucketize-procedure.js [hunk 2]
```diff
@@ -62,6 +62,11 @@ function runTest(testNumber, buckets, goldStandard) {
     assertEqual(res['responseCode'], 200);
     checkResults(res, goldStandard);
 
+    res = mldb.get("/v1/query", {q: "SELECT *, when({*}) FROM " + datasetName,
+                                 format: "full"});
+
+    // MLDB-1267
+    assertEqual(res.json[0].columns[1][1].ts, "2015-12-22T14:26:51Z");
```
Comment (guyd) [inline testing/MLDB-1043-bucketize-procedure.js]: Wouldn't this be dependent on when the check-out was made?

File: plugins/bucketize_procedure.cc [hunk 3]
```diff
@@ -143,26 +161,29 @@ run(const ProcedureRunConfig & run,
...
                                 nullptr, true /*overwrite*/);
 
-    typedef tuple<ColumnName, CellValue, Date> cell;
-    PerThreadAccumulator<vector<pair<RowName, vector<cell>>>> accum;
+    typedef tuple<ColumnName, CellValue, Date> Cell;
+    PerThreadAccumulator<vector<pair<RowName, vector<Cell> > > > accum;
 
     for (const auto & mappedRange: procedureConfig.percentileBuckets) {
-        string bucketName(mappedRange.first);
+        ColumnName bucketColumn("bucket");
```
Comment (FinchPowers) [inline plugins/bucketize_procedure.cc]: This variable is unused.

File: plugins/bucketize_procedure.cc [hunk 4]
```diff
@@ -177,14 +197,14 @@ run(const ProcedureRunConfig & run,
...
-             << " to " << higherBound << endl;
+        //cerr << "Bucket " << mappedRange.first << " from " << lowerBound
+        //     << " to " << higherBound << endl;
 
         ML::run_in_parallel_blocked(lowerBound, higherBound, applyFct);
     }
 
     // record remainder
-    accum.forEach([&] (vector<pair<RowName, vector<cell>>> * rows)
+    accum.forEach([&] (vector<pair<RowName, vector<Cell> > > * rows)
```
Comment (FinchPowers) [inline plugins/bucketize_procedure.cc]: Why the extra whitespaces? Is it auto formatting? I think old c++ compilers required it, is that the reason?

File: testing/MLDB-1043-bucketize-procedure.js (hunk 2 above)
Comment (FinchPowers) [inline testing/MLDB-1043-bucketize-procedure.js]: This is indeed a bug.

**Pull request merged**